jax/_src/cudnn/fused_attention_stablehlo.py

```diff
@@ -100,50 +103,51 @@ def create_dot_product_attention_backend_config(batch,
     "is_flash_attention": is_flash_attention,
     "is_causal_mask": is_causal_mask,
   }
-  fwd_dot_number = {
-    "bmm1_dot_dimension_numbers": {
-      "lhs_contracting_dimensions": ["3"],
-      "rhs_contracting_dimensions": ["3"],
-      "lhs_batch_dimensions": ["0", "2"],
-      "rhs_batch_dimensions": ["0", "2"],
-    },
-    "bmm2_dot_dimension_numbers": {
-      "lhs_contracting_dimensions": ["3"],
-      "rhs_contracting_dimensions": ["1"],
-      "lhs_batch_dimensions": ["0", "1"],
-      "rhs_batch_dimensions": ["0", "2"],
-    },
-  }
-  bwd_dot_number = {
-    "bmm1_grad_gemm1_dot_dimension_numbers": {
-      "lhs_contracting_dimensions": ["2"],
-      "rhs_contracting_dimensions": ["1"],
-      "lhs_batch_dimensions": ["0", "1"],
-      "rhs_batch_dimensions": ["0", "2"],
-    },
-    "bmm1_grad_gemm2_dot_dimension_numbers": {
-      "lhs_contracting_dimensions": ["3"],
-      "rhs_contracting_dimensions": ["1"],
-      "lhs_batch_dimensions": ["0", "1"],
-      "rhs_batch_dimensions": ["0", "2"],
-    },
-    "bmm2_grad_gemm1_dot_dimension_numbers": {
-      "lhs_contracting_dimensions": ["2"],
-      "rhs_contracting_dimensions": ["1"],
-      "lhs_batch_dimensions": ["0", "1"],
-      "rhs_batch_dimensions": ["0", "2"],
-    },
-    "bmm2_grad_gemm2_dot_dimension_numbers": {
-      "lhs_contracting_dimensions": ["3"],
-      "rhs_contracting_dimensions": ["3"],
-      "lhs_batch_dimensions": ["0", "2"],
-      "rhs_batch_dimensions": ["0", "2"],
-    },
-  }
+  
+  # We define the contracting and batch dims in the format of
+  # ((lhs_contracting_dims, rhs_contracting_dims), (lhs_batch_dims,
+  # rhs_batch_dims)).
+  if layout == 'BNTH':
+    dims = [
+        ((3, 3), ((0, 1), (0, 1))), # BMM1: BNTH,BNSH->BNTS
+        ((3, 2), ((0, 1), (0, 1))), # BMM2: BNTS,BNSH->BNTH
+        ((2, 2), ((0, 1), (0, 1))), # BMM1_grad_1: BNTS,BNTH->BNSH
+        ((3, 2), ((0, 1), (0, 1))), # BMM1_grad_2: BNTS,BNSH->BNTH
+        ((2, 2), ((0, 1), (0, 1))), # BMM2_grad_1: BNTS,BNTH->BNSH
+        ((3, 3), ((0, 1), (0, 1))), # BMM2_grad_2: BNTH,BNSH->BNTS
+    ]
+  else:
+    dims = [
+        ((3, 3), ((0, 2), (0, 2))), # BMM1: BTNH,BSNH->BNTS
+        ((3, 1), ((0, 1), (0, 2))), # BMM2: BNTS,BSNH->BTNH
+        ((2, 1), ((0, 1), (0, 2))), # BMM1_grad_1: BNTS,BTNH->BSNH
+        ((3, 1), ((0, 1), (0, 2))), # BMM1_grad_2: BNTS,BSNH->BTNH
+        ((2, 1), ((0, 1), (0, 2))), # BMM2_grad_1: BNTS,BTNH->BSNH
+        ((3, 3), ((0, 2), (0, 2))), # BMM2_grad_2: BTNH,BSNH->BNTS
+    ]
+  keys = [
+      "bmm1_dot_dimension_numbers",
```

Comment:
why not just merge dim and keys into a dict

Reply:
I think this is mainly for format and readability. Putting the comment besides the dims is more informative, I think.